packages/in17n/src/lib/In17nHandler.ts

```diff
@@ -0,0 +1,145 @@
+		this.options = options;
+		this.languagesDir = this.options?.defaultLanguageDirectory ?? join(getRootData().root, 'languages');
+
+		this.backendOptions = mergeDefault(
+			{
+				loadPath: join(this.languagesDir, '{{lng}}', '{{ns}}.json'),
+				addPath: this.languagesDir
+			} as i18nextFsBackend.i18nextFsBackendOptions,
+			this.options?.backend
+		);
+	}
+
+	/**
+	 * Intitialises the handler by loading in the namespaces, passing the data to i18next, and filling in the {@link In17nHandler#languages}.
+	 * @since 1.0.0
+	 */
+	public async init() {
+		const { namespaces, languages } = await this.walkLanguageDirectory(this.languagesDir);
+
+		i18next.use(Backend);
+		await i18next.init(
+			mergeDefault(
+				{
+					backend: this.backendOptions,
+					fallbackLng: this.options?.defaultName ?? 'en-US',
+					initImmediate: false,
+					interpolation: {
+						escapeValue: false
+					},
+					load: 'all',
+					defaultNS: this.options?.defaultNS ?? 'default',
+					ns: namespaces,
+					preload: languages
+				} as InitOptions,
+				this.options?.i18next
+			)
+		);
+
+		this.languages = new Map(languages.map((item) => [item, i18next.getFixedT(item)]));
+		this.languagesLoaded = true;
+	}
+
+	/**
+	 * Retrieve a raw TFunction from the passed locale.
+	 * @param locale The language to be used.
+	 * @since 1.0.0
+	 */
+	public fetchT(locale: string) {
+		if (!this.languagesLoaded) throw new Error('Cannot call this method until In17nHandler#init has been called');
+
+		const t = this.languages.get(locale);
+		if (!t) throw new ReferenceError('Invalid language provided');
+
+		return t;
+	}
+
+	/**
+	 * Resolves a localised string from a language code, key, optional replaceables, and optional i18next options.
+	 * @param locale The language to be used.
+	 * @param key The key that should be translated.
+	 * @param replace The replaceable keys in translation string.
+	 * @param options i18next language options.
+	 * @since 1.0.0
+	 */
+	public fetchLocale(locale: string, key: string, replace?: Record<string, unknown>, options?: TOptions<StringMap>): Awaited<string> {
+		if (!this.languagesLoaded) throw new Error('Cannot call this method until In17nHandler#init has been called');
+
+		const language = this.languages.get(locale);
+		if (!language) throw new ReferenceError('Invalid language provided');
+
+		return language(
+			key,
+			mergeDefault(
+				{
+					defaultValue: language(this.options?.defaultMissingKey ?? 'default:default', { replace: { key } }),
+					replace
+				},
+				options
+			)
+		);
+	}
+
+	/**
+	 * @description Skips any files that don't end with `.json`.
+	 * @param dir The directory that should be walked.
+	 * @param namespaces The currently known namespaces.
+	 * @param folderName The currently walked folder.
+	 * @since 1.0.0
+	 * @protected
+	 */
+	protected async walkLanguageDirectory(dir: string, namespaces: string[] = [], folderName = '') {
+		const files = await readdir(dir);
```

Comment:
Use `fs/promises.opendir`, it's more convenient, uses less resources, and returns `Dirent`s, which can tell you whether something is a file, a directory, or any other thing, without extra calls.

Also uses for-await-of, which is pretty convenient as it allows for zero-copy operation.